File: src/main/java/upbrella/be/store/dto/response/AllStoreIntroductionResponse.java
package upbrella.be.store.dto.response;

import lombok.Builder;
import lombok.Getter;

import java.util.List;

@Getter
@Builder
public class AllStoreIntroductionResponse {


    private List<StoreIntroductionsResponseByClassification> storesByClassification;

    public static AllStoreIntroductionResponse of(List<StoreIntroductionsResponseByClassification> storesByClassification) {

        return AllStoreIntroductionResponse.builder()
                .storesByClassification(storesByClassification)
                .build();
    }
}

File: src/main/kotlin/upbrella/be/store/dto/response/AllStoreIntroductionResponse.kt
package upbrella.be.store.dto.response

data class AllStoreIntroductionResponse(
    val storesByClassification: List<StoreIntroductionsResponseByClassification>
) {
    companion object {
        fun of(storesByClassification: List<StoreIntroductionsResponseByClassification>) =
            AllStoreIntroductionResponse(storesByClassification)
    }
}
